### PR TITLE
docs: typedoc automated SDK documentation

### PR DESCRIPTION
**Pull request by dmulvi**
## Description

wip: this is the start of automating the documentation of our SDKs

## Test plan

testing locally

### CODE REVIEW

File: packages/client/wallets/smart-wallet/custom-plugin.mjs
Comment (devlyn37) [inline packages/client/wallets/smart-wallet/custom-plugin.mjs:1]: I'd guess most of this isn't smart wallet specific and will eventually go to some shared spot like the `@crossmint/sdk-common` right?

**Review by devlyn37 — APPROVED** [review]
LGTM, left a few comments and questions but non-blocking. Approving assuming you'll fix whatever is breaking the build here!

File: packages/client/wallets/smart-wallet/tsconfig.typedoc.json
[the remark this answers is not in the document]
Comment (dmulvi) [reply packages/client/wallets/smart-wallet/tsconfig.typedoc.json:1]: good call!

File: packages/client/wallets/smart-wallet/custom-plugin.mjs
Comment (dmulvi) [reply packages/client/wallets/smart-wallet/custom-plugin.mjs:1]: Correct, most of it is not smart wallet specific. For now keeping as-is until I can make the time to look into pushing some of this code up a few levels in the SDK. That will require some more testing, at which point I'll fix here as well.

**Pull request merged**